### PR TITLE
Adds to and toMat to SourceWithContext

### PR DESCRIPTION
**Pull request by skyluc**
Updated tests to use the new functions.

### CODE REVIEW

**Review by johanandren — CHANGES_REQUESTED** [review]
Looks good but missing the corresponding signatures in `akka.stream.javadsl.SourceWithContext`

**Review by johanandren — APPROVED** [review]
LGTM

File: akka-stream-tests/src/test/scala/akka/stream/scaladsl/SourceWithContextSpec.scala [hunk 1]
```diff
@@ -112,8 +112,8 @@ class SourceWithContextSpec extends StreamSpec {
         .startContextPropagation(_.offset)
         .map(_.data)
         .statefulMapConcat(statefulFunction)
-        .endContextPropagation
-        .runWith(TestSink.probe[(String, Long)])
+        .toMat(TestSink.probe[(String, Long)])(Keep.right)
+        .run
```
Comment (patriknw) [inline akka-stream-tests/src/test/scala/akka/stream/scaladsl/SourceWithContextSpec.scala]: shouldn't `SourceWithContext` also have `runWith` for consistency with ordinary `Source`, making this typical case more convenient

**Comment by patriknw** [comment]
Cont in https://github.com/akka/akka/pull/26464

**Pull request closed without merging**